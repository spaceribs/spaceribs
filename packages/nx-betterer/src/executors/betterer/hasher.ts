import { CustomHasher, Hash } from '@nrwl/devkit';
import { resolve } from 'path';

export const bettererHasher: CustomHasher = async (task, context) => {
  const hashTask = await context.hasher.hashTask(task);
  const bettererCachePath = resolve('tmp', task.projectRoot, '.betterer.cache');
  const hashFile = context.hasher.hashFile(bettererCachePath);

  const hash: Hash = {
    value: hashFile,
    details: hashTask.details,
  };

  return Promise.resolve(hash);
};

export default bettererHasher;
