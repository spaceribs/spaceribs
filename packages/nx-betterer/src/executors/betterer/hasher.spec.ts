import { Hasher, HasherContext } from '@nrwl/devkit';

import { bettererHasher } from './hasher';

describe('bettererHasher', () => {
  it('should generate hash', async () => {
    const mockHasher: Hasher = {
      hashFile: jest.fn().mockReturnValue('test123'),
      hashTask: jest
        .fn()
        .mockReturnValue({ value: 'hashed-task', details: { test: 123 } }),
    } as unknown as Hasher;

    const hash = await bettererHasher(
      {
        projectRoot: 'root',
        id: 'my-task-id',
        target: {
          project: 'proj',
          target: 'target',
        },
        overrides: {},
      },
      {
        hasher: mockHasher,
        projectName: 'proj',
        workspace: {
          projects: {
            proj: {
              root: 'libs/proj',
            },
          },
        },
      } as unknown as HasherContext
    );
    expect(hash).toMatchSnapshot();
  });
});
