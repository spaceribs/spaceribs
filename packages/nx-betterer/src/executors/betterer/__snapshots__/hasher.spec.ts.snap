// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bettererHasher should generate hash 1`] = `
Object {
  "details": Object {
    "test": 123,
  },
  "value": "test123",
}
`;
